src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -435,10 +440,10 @@ def _request_dataloader(self, stage: RunningStage) -> Union[DataLoader, List[Dat
         return dataloader
 
     @staticmethod
-    def _resolve_overfit_batches(dataloaders: Collection[DataLoader], mode: RunningStage) -> Collection[DataLoader]:
+    def _resolve_overfit_batches(dataloaders: BATCH_DATALOADER, mode: RunningStage) -> BATCH_DATALOADER:
```

Comment:
```suggestion
    def _resolve_overfit_batches(dataloaders: EVAL_DATALOADERS, mode: RunningStage) -> EVAL_DATALOADERS:
```
from `utilities/types.py`

Reply:
combining these suggestions with the other suggestions also causes additional errors